File: Raven.Client.WinRT/Connection/ReplicationInformerLocalCache.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Raven.Abstractions.Data;
using Raven.Abstractions.Extensions;
using Raven.Abstractions.Logging;
using Windows.Storage;

namespace Raven.Client.Connection
{
	public static class ReplicationInformerLocalCache
	{
		private readonly static ILog log = LogManager.GetCurrentClassLogger();
		private static readonly StorageFolder folder = ApplicationData.Current.RoamingFolder;

		public static JsonDocument TryLoadReplicationInformationFromLocalCache(string serverHash)
		{
			try
			{
				return Load(serverHash).Result;
			}
			catch (Exception e)
			{
				log.ErrorException("Could not understand the persisted replication information", e);
				return null;
			}
		}

		private static async Task<JsonDocument> Load(string serverHash)
		{
			var path = "RavenDB Replication Information For - " + serverHash;

			StorageFile file;
			try
			{
				file = await folder.GetFileAsync(path);
			}
			catch (FileNotFoundException)
			{
				return null;
			}

			using (var stream = await file.OpenSequentialReadAsync())
			{
				return stream.AsStreamForRead().ToJObject().ToJsonDocument();
			}
		}

		public static void TrySavingReplicationInformationToLocalCache(string serverHash, JsonDocument document)
		{
			try
			{
				Save(serverHash, document).Wait();
			}
			catch (Exception e)
			{
				log.ErrorException("Could not persist the replication information", e);
			}
		}

		private static async Task Save(string serverHash, JsonDocument document)
		{
			var path = "RavenDB Replication Information For - " + serverHash;

			var file = await folder.CreateFileAsync(path, CreationCollisionOption.ReplaceExisting);
			using (var stream = await file.OpenStreamForWriteAsync())
			{
				document.ToJson().WriteTo(stream);
			}
		}
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/connection/ReplicationInformerLocalCache.java
package net.ravendb.client.connection;

import java.io.File;

import net.ravendb.abstractions.data.JsonDocument;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.abstractions.logging.ILog;
import net.ravendb.abstractions.logging.LogManager;

import org.apache.commons.io.FileUtils;
import org.apache.commons.lang.StringUtils;


public class ReplicationInformerLocalCache {

  private static ILog log = LogManager.getCurrentClassLogger();

  private static String tempDir = System.getProperty("java.io.tmpdir");

  public static JsonDocument tryLoadReplicationInformationFromLocalCache(String serverHash) {
    JsonDocument result = null;
    try {
      String path = "RavenDB Replication Information For - " + serverHash;
      File file = new File(tempDir, path);
      String fileContent = FileUtils.readFileToString(file);
      if (StringUtils.isBlank(fileContent)) {
        return null;
      }
      result = SerializationHelper.toJsonDocument(RavenJObject.parse(fileContent));
    } catch (Exception e) {
      log.error("Could not understand the persisted replication information", e);
      return null;
    }
    return result;
  }

  public static void trySavingReplicationInformationToLocalCache(String serverHash, JsonDocument document) {
    try {
      String path = "RavenDB Replication Information For - " + serverHash;
      File file = new File(tempDir, path);
      FileUtils.writeStringToFile(file, document.toJson().toString());
    } catch (Exception e) {
      log.error("Could not persist the replication information", e);
    }
  }

}
